Rewrite conclusion (#62)
greenelab/text_mined_hetnet_manuscript: content/05.discussion_and_conclusion.md

## Discussion and Conclusions

Filling out knowledge bases via manual curation can be an arduous and erroneous task [@doi:10.1093/bioinformatics/btm229].
Using manual curation alone becomes impractical as the rate of publications continuously increases.
Data programming is a paradigm that uses label functions to speed up the annotation process and can be used to solve this problem.
However, creating useful label functions is an obstacle to this paradigm, which takes considerable time.
We tested the feasibility of re-using label functions to reduce the number of label functions required for strong prediction performance.

Our sampling experiment revealed that adding edge-specific label functions is better than adding off-edge label functions.
An exception to this trend is using label functions designed from conceptually related edge types (using Gene-interacts-Gene (GiG) label functions to predict Compound-binds-Gene (CbG) sentences and vice versa).

> Furthermore, broad edge types such as Disease-associates-Gene (DaG) did not follow this trend as we found this edge to be agnostic to any tested label function source.

Reviewer: What do you mean by agnostic? Didn't benefit from any label functions including its own?

Author: The opposite in this case. Performance increased regardless of the label function source. So agnostic in this case means that the edge type didn't prefer one source over the others.

One possibility for this observation is that the "associates" relationship is a general concept that may include other concepts such as Disease (up/down) regulating a Gene (examples highlighted in our [annotated sentences](https://github.com/greenelab/text_mined_hetnet_manuscript/tree/master/supplementary_materials/annotated_sentences)).
The discriminator model did not have an apparent positive or negative effect on performance; however, we noticed that performance heavily depended on the annotations provided by the generative model.

> This pattern suggests that future endeavors should emphasize label function construction and generative model training over the discriminative model to achieve better downstream performance.

Reviewer: ```suggestion
This pattern suggests a focus on label function construction and generative model training may be key steps to focus on in future work.
```

> Although our conclusion states that label functions cannot be re-used across all edge types, other strategies like multitask learning [@doi:10.1145/3209889.3209898] or transfer learning [@doi:10.1186/s40537-016-0043-6] could make mining multiple edge types more practical.

Reviewer: ```suggestion
Although we found that label functions cannot be re-used across all edge types with the standard task framing, strategies like multitask [@doi:10.1145/3209889.3209898] or transfer learning [@doi:10.1186/s40537-016-0043-6] may make multi-label-function efforts more successful.
```

## Discussion and Conclusions

Filling out knowledge bases via manual curation can be an arduous and erroneous task [@doi:10.1093/bioinformatics/btm229].
Using manual curation alone becomes impractical as the rate of publications continuously increases.
Data programming is a paradigm that uses label functions to speed up the annotation process and can be used to solve this problem.
However, creating useful label functions is an obstacle to this paradigm, which takes considerable time.
We tested the feasibility of re-using label functions to reduce the number of label functions required for strong prediction performance.

> Through our sampling experiment, we found that adding edge-specific label functions increases performance for the generative model (Figure {@fig:auroc_gen_model_test_set}).

Reviewer: ```suggestion
Our sampling experiment revealed that adding edge-specific label functions increases performance for the generative model.
```
If one is citing figures here, it's probably too detailed. I recommend citing figures almost exclusively within the results.

> We found that label functions designed from relatively related edge types can increase performance (Gene-interacts0Gene (GiG) label functions predicting the Compound-binds-Gene (CbG) edge and vice versa), while the Disease-associates-Gene (DaG) edge type remained agnostic to label function sources (Figure {@fig:auroc_gen_model_test_set} and Supplemental Figure {@fig:aupr_gen_model_test_set}).

Reviewer: ```suggestion
We found that label functions designed from conceptually related edge types can increase performance (Gene-interacts-Gene (GiG) label functions predicting the Compound-binds-Gene (CbG) edge and vice versa), while others did not.
```

> Furthermore, we found that using all label functions at once generally hurts performance with the exception being the DaG edge type (Supplemental Figures {@fig:auroc_grabbag_gen_model_test_set} and {@fig:aupr_grabbag_gen_model_test_set}).

Reviewer: ```suggestion
Using all label functions at once generally reduced performance with the exception being the DaG edge type.
```

> One possibility for this observation is that DaG is a broadly defined edge type.

Reviewer: ```suggestion
One possibility for this observation is that the "associates" relationship is a general concept.
```

> For example, DaG may contain many concepts related to other edge types, such as Disease (up/down) regulating a Gene, which makes it more agnostic to label function sources (examples highlighted in our [annotated sentences](https://github.com/greenelab/text_mined_hetnet_manuscript/tree/master/supplementary_materials/annotated_sentences)).  

Reviewer: ```suggestion
For example, DaG may contain many concepts related to other edge types, such as a Disease (up/down) regulating a Gene, which makes it more agnostic to label function sources (examples highlighted in our [annotated sentences](https://github.com/greenelab/text_mined_hetnet_manuscript/tree/master/supplementary_materials/annotated_sentences)).  
```

> Regarding the discriminative model, adding edge-specific label function improved performance for two out of the four edge types (Compound-treats-Disease (CtD) and  DaG) (Figure {@fig:auroc_discriminative_model_performance} and Supplemental Figure {@fig:aupr_discriminative_model_performance}). 

Reviewer: Could this entire paragraph be summarized as: "In this context, the discriminative model did not have a clear positive or negative effect on performance."

I am worried this is so detailed that it's restating the results.

In terms of CbG, the discriminator model failed to improve over the generative model (Figure {@fig:auroc_discriminative_model_performance} and Supplemental Figure {@fig:aupr_discriminative_model_performance}).
A possible explanation for hindered performance is that these edge types contain many sentences with spurious gene mentions, resulting in many false-positive examples.
Improving performance for all edge types would require more labeled examples for the tuning set, along with the construction of more label functions for the generative model.
Even with these limitations, we estimated the recall amount of edges within Hetionet v1.
We found that our approach could recall a modest amount of existing edges while proposing a significant amount of new edges to be included (Figure {@fig:hetionet_reconstruction}). 
These results suggest that text mining is a viable approach to expanding existing knowledge bases; however, more advanced techniques [@doi:10.1145/2623330.2623623] may be necessary to combat the problem of high false positives.

> Overall, we conclude that the re-use of label functions is possible for closely related edge types, but it does not improve performance for most pairings.

Reviewer: ```suggestion
Overall, we conclude that the re-use of label functions may improve performance for closely related edge types but does not improve performance for most pairings.
```
